component/attr_cache/attr_cache.go

```diff
@@ -752,7 +708,20 @@ func (ac *AttrCache) WriteFile(options internal.WriteFileOptions) (int, error) {
 		ac.cacheLock.RLock()
 		defer ac.cacheLock.RUnlock()
 		// TODO: Could we just update the size and mod time of the file here? Or can other attributes change here?
-		ac.invalidatePath(options.Handle.Path)
+
+		var toBeInvalid *attrCacheItem
+		var err error
+		if attr == nil {
+			toBeInvalid, err = ac.cacheMap.get(options.Handle.Path)
+		} else {
+			toBeInvalid, err = ac.cacheMap.get(attr.Path) //attr is nil here for TestWriteFileDoesNotExist()
+		}
+
+		if err != nil {
+			log.Err("could not find attribute item in cache to invalidate due to the following error: ", err)
+		} else {
+			toBeInvalid.invalidate()
+		}
 	}
 	return size, err
```

Comment:
The error coming back from the call to `get()` needs to have a different name so the function doesn't end up returning the `get()` error when it should return nil.
This comes up often enough in the code that it might be a good idea to scour the code for this pattern.